src/ln/channelmanager.rs

```diff
@@ -391,6 +391,20 @@ pub struct ChannelDetails {
 	pub channel_value_satoshis: u64,
 	/// The user_id passed in to create_channel, or 0 if the channel was inbound.
 	pub user_id: u64,
+	/// The available outbound capacity for sending HTLCs to the remote peer. This does not include
+	/// any pending HTLCs which are not yet fully resolved (and, thus, who's balance is not
+	/// available for inclusion in new outbound HTLCs). This further does not include any pending
+	/// outgoing HTLCs which are awaiting some other resolution to be sent.
+	pub outbound_capacity_msat: u64,
+	/// The available inbound capacity for the remote peer to send HTLCs to us. This does not
+	/// include any pending HTLCs which are not yet fully resolved (and, thus, who's balance is not
+	/// available for inclusion in new inbound HTLCs).
+	/// Note that there are some corner cases not fully handled here, so the actual available
+	/// inbound capacity may be slightly higher than this.
+	pub inbound_capacity_msat: u64,
+	/// True if the channel is (a) confirmed and funding_locked messages have been exchanged, (b)
```

Comment:
What's about channel_update with disable flags set ? Or if we detect channel transaction but closing hasn't get enough confs (and so channel isn't prune yet from channel graph) ?

Reply:
As for (a), we handle that in Router, and I'm still pretty confused as to how a peer can send that to you and have you do anything, do we implement it yet anyway?, as for (b) I think we prune it immediately, no?

Reply:
Okay, so "liveness" is just a ChannelManager view-based and doesn't rely on what the Router may receive (like a channel_update from peer disabling this channel, which is not implemented yet because #207 is still on stand-by)

Reply:
I *think* #207 should result in the Channel knowing if the counterparty sent us a channel_update disabling our own channel, then we could view it, but, yes.